clang/lib/Sema/SemaSYCLDeclAttr.cpp

```diff
@@ -132,6 +132,13 @@ void SemaSYCL::checkDeprecatedSYCLAttributeSpelling(const ParsedAttr &A,
     return;
   }
 
+  // Additionally, diagnose deprecated [[sycl::reqd_sub_group_size]] spelling
+  if (A.getKind() == ParsedAttr::AT_IntelReqdSubGroupSize &&
+      A.getAttrName()->isStr("reqd_sub_group_size") && A.getScopeName()->isStr("intel")) {
```

Comment:
Do we really need to check the name here? I would expect it to be covered by `kind` check already

Reply:
No need, fixed